core/chains/evm/logpoller/parser.go

```diff
@@ -0,0 +1,499 @@
+	)
+}
+
+func (v *pgDSLParser) Confirmations(p primitives.Confirmations) {
+	switch p.ConfirmationLevel {
+	case primitives.Finalized:
+		v.expression = v.nestedConfQuery(true, 0)
+	case primitives.Unconfirmed:
+		// Unconfirmed in the evm relayer is an alias to the case of 0 confirmations
+		// set the level to the number 0 and fallthrough to the default case
+		p.ConfirmationLevel = primitives.ConfirmationLevel(0)
+
+		fallthrough
+	default:
+		// the default case passes the confirmation level as a number directly to a subquery
+		v.expression = v.nestedConfQuery(false, uint64(evmtypes.Confirmations(p.ConfirmationLevel)))
+	}
+}
+
+func (v *pgDSLParser) Timestamp(p primitives.Timestamp) {
+	cmp, err := cmpOpToString(p.Operator)
+	if err != nil {
+		v.err = err
+
+		return
+	}
+
+	v.expression = fmt.Sprintf(
+		"%s %s :%s",
+		timestampFieldName,
+		cmp,
+		v.args.withIndexedField(timestampFieldName, time.Unix(int64(p.Timestamp), 0)),
+	)
+}
+
+func (v *pgDSLParser) TxHash(p primitives.TxHash) {
+	bts, err := hexutil.Decode(p.TxHash)
+	if errors.Is(err, hexutil.ErrMissingPrefix) {
+		bts, err = hexutil.Decode("0x" + p.TxHash)
+	}
+
+	if err != nil {
+		v.err = err
+
+		return
+	}
+
+	txHash := common.BytesToHash(bts)
+
+	v.expression = fmt.Sprintf(
+		"%s = :%s",
+		txHashFieldName,
+		v.args.withIndexedField(txHashFieldName, txHash),
+	)
+}
+
+func (v *pgDSLParser) VisitAddressFilter(p *addressFilter) {
+	v.expression = fmt.Sprintf(
+		"address = :%s",
+		v.args.withIndexedField("address", p.address),
+	)
+}
+
+func (v *pgDSLParser) VisitEventSigFilter(p *eventSigFilter) {
+	v.expression = fmt.Sprintf(
+		"%s = :%s",
+		eventSigFieldName,
+		v.args.withIndexedField(eventSigFieldName, p.eventSig),
+	)
+}
+
+func (v *pgDSLParser) nestedConfQuery(finalized bool, confs uint64) string {
+	var (
+		from     = "FROM evm.log_poller_blocks "
+		where    = "WHERE evm_chain_id = :evm_chain_id "
+		order    = "ORDER BY block_number DESC LIMIT 1)"
+		selector string
+	)
+
+	if finalized {
+		selector = "(SELECT finalized_block_number "
+	} else {
+		selector = fmt.Sprintf("(SELECT greatest(block_number - :%s, 0) ",
+			v.args.withIndexedField("confs", confs),
+		)
+	}
+
+	var builder strings.Builder
+
+	builder.WriteString(selector)
+	builder.WriteString(from)
+	builder.WriteString(where)
+	builder.WriteString(order)
+
+	return fmt.Sprintf("%s <= %s", blockFieldName, builder.String())
+}
+
+func (v *pgDSLParser) VisitEventByWordFilter(p *eventByWordFilter) {
+	if len(p.ValueComparers) > 0 {
+		wordIdx := v.args.withIndexedField("word_index", p.WordIndex)
+
+		comps := make([]string, len(p.ValueComparers))
+		for idx, comp := range p.ValueComparers {
+			comps[idx], v.err = makeComp(comp, v.args, "word_value", wordIdx, "substring(data from 32*:%s+1 for 32) %s :%s")
+			if v.err != nil {
+				return
+			}
+		}
+
+		v.expression = strings.Join(comps, " AND ")
+	}
+}
+
+func (v *pgDSLParser) VisitEventTopicsByValueFilter(p *eventByTopicFilter) {
```

Comment:
It seems that `p.EventSig` is completely ignored here

Reply:
This needs to be removed.